system-parachains/bridge-hubs/bridge-hub-kusama/src/xcm_config.rs

```diff
@@ -342,3 +371,25 @@ fn treasury_pallet_account_not_none() {
 		LocationToAccountId::convert_location(&RelayTreasuryLocation::get()).unwrap()
 	)
 }
+
+/// A `FeeManager` implementation that forces fees for any message delivered to Ethereum.
+/// Otherwise, it permits the specified `WaivedLocations` to not pay for fees and uses the provided
+/// `HandleFee` implementation.
+pub struct XcmFeeManagerFromComponentsBridgeHub<WaivedLocations, HandleFee>(
+	PhantomData<(WaivedLocations, HandleFee)>,
+);
+impl<WaivedLocations: Contains<Location>, FeeHandler: HandleFee> FeeManager
+	for XcmFeeManagerFromComponentsBridgeHub<WaivedLocations, FeeHandler>
+{
+	fn is_waived(origin: Option<&Location>, fee_reason: FeeReason) -> bool {
+		let Some(loc) = origin else { return false };
+		if let Export { network, destination: Here } = fee_reason {
+			return !(network == EthereumNetwork::get())
```

Comment:
should this not still check `WaivedLocations` when exporting to something else than Ethereum? aka, only short-circuit return false for Ethereum, but still check for all other cases.

(same for BHP)

Reply:
Yep, it is checked here 2 lines down: https://github.com/polkadot-fellows/runtimes/pull/130/files/bd2c69f2c31f289da11c41647deb1d23938c94a1#diff-9c700431fc38984083e2309b01d164c9b15bcbedd02a96210e73ea9265cd5f7eR389

So the behaviour is the same as `XcmFeeManagerFromComponents` except when exporting to Ethereum.

Reply:
my point is that it's not equivalent, it shortcircuits and returns true or false without checking WaivedLocations.

e.g. `fee_reason = Export { network: Polkadot, destination: Here }` would just (incorrectly) return `true`

I was thinking changing this to smth like:

```rust
	fn is_waived(origin: Option<&Location>, fee_reason: FeeReason) -> bool {
		let Some(loc) = origin else { return false };
		if let Export { network, destination: Here } = fee_reason {
			if network == EthereumNetwork::get() {
				return false
			}
		}
		WaivedLocations::contains(loc)
	}
```

Reply:
Great catch, thanks @acatangiu. I changed it and added some unit tests to check the various permutations. https://github.com/polkadot-fellows/runtimes/pull/130/commits/336acb042e7aa6d0cb76681dfc25eb59333795b2

Reply:
looks great, thanks for the good test coverage! 💪